tests/e2e/utils/DriverHelper.ts

```diff
@@ -194,99 +202,121 @@ export class DriverHelper {
     }
 
     public async waitAllDisappearance(locators: Array<By>,
-        attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
-        polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
+        attemptsPerLocator: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
+        pollingPerLocator: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
 
         Logger.trace(`DriverHelper.waitAllDisappearance ${locators}`);
 
         for (const elementLocator of locators) {
-            await this.waitDisappearance(elementLocator, attempts, polling);
+            await this.waitDisappearance(elementLocator, attemptsPerLocator, pollingPerLocator);
         }
     }
 
     public async waitAndClick(elementLocator: By, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
         const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
 
         Logger.trace(`DriverHelper.waitAndClick ${elementLocator}`);
 
         for (let i = 0; i < attempts; i++) {
-            const element: WebElement = await this.waitVisibility(elementLocator, timeout);
-
             try {
```

Comment:
2 levels of try/catch look redundant here and lead to increase the complexity of the code

Reply:
I had issues for the methods failing after times that did not match the timeout without a sensual exception message.
The waitVisibility method was also failing with ElementNotFound that was never caught and caused issue with the logic of the tests.
The second layer of try-catch is there to prevent the method failing without reaching the desired amount of attempts/timeout and if the issue persists, it throws the correct exception.